iothub/service/src/DigitalTwin/DigitalTwinClient.cs

```diff
@@ -36,7 +40,40 @@ public static DigitalTwinClient CreateFromConnectionString(string connectionStri
 
         private DigitalTwinClient(Uri uri, IotServiceClientCredentials credentials, params DelegatingHandler[] handlers)
         {
-            _client = new IotHubGatewayServiceAPIs(uri, credentials, handlers);
+            // ConnectionLeaseTimeoutHandler is IDisposable, so save it as a field so it can be disposed later.
+            _connectionLeaseTimeoutHandler = new ConnectionLeaseTimeoutHandler(HttpTransportSettings.DefaultConnectionLeaseTimeoutMilliseconds);
+
+            // Assign each custom delegating handler as the inner handler of the previous delegating handler
+            // For instance:
+            // _connectionLeaseTimeoutHandler.InnerHandler = handlers[0],
+            // handlers[0].InnerHandler = handlers[1],
+            // handlers[1].InnerHandler = handlers[2],
+            // ...
+            // handlers[handlers.Length].InnerHandler = _httpClientHandler
+            DelegatingHandler chainedDelegatingHandler = _connectionLeaseTimeoutHandler;
+            foreach (DelegatingHandler delegatingHandler in handlers)
+            {
+                chainedDelegatingHandler.InnerHandler = delegatingHandler;
```

Comment:
more questions coming to mind = what if the input is a delegating handler with an inner handler set? we'd then be overwriting it here. maybe some kind of check here could be helpful.

Reply:
I'll be removing this chaining and passing down the delegating handler list + the one new connection lease timeout handler directly to the generated code, so this won't be an issue anymore